src/MongoDB.Driver.Legacy/Builders/IndexKeysBuilder.cs

```diff
@@ -261,6 +271,30 @@ public override BsonDocument ToBsonDocument()
             return _document;
         }
 
+        /// <summary>
+        /// Sets a wildcard key to the index. If the wildcard name is empty, the generated key will be for the whole collection.
+        /// </summary>
+        /// <param name="name">The wildcard key.</param>
+        /// <returns>The builder (so method calls can be chained).</returns>
+        public IndexKeysBuilder Wildcard(string name = null)
+        {
+            string wildcard = name;
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                wildcard = "$**";
+            }
+            else
+            {
+                if (!name.EndsWith("$**"))
```

Comment:
It seems odd to check name and modify wildcard.

Perhaps check wildcard instead?

I know they have the same value... but I have to examine the whole method to convince myself of that. That's a consequence of having two variables.

Reply:
Done